### PR TITLE
Simplified configuration

### PR DESCRIPTION
**Pull request by hvge**
This PR adds support for simplified cryptographic configuration.

### CODE REVIEW

File: include/PowerAuth/Session.h [hunk 1]
```diff
@@ -78,12 +78,12 @@ namespace powerAuth
          session has a no valid setup.
          */
         const SessionSetup * sessionSetup() const;
-        
         /**
-         Returns value of sessionSetup()->sessionIdentifier if the setup structure is present or 0 if not.
+         Returns application key extracted from internal SessionSetup, or empty string
+         if session has a no valid setup.
```
Comment (kober32) [inline include/PowerAuth/Session.h:83]: Typo

File: include/PowerAuth/Session.h [hunk 2]
```diff
@@ -78,12 +78,12 @@ namespace powerAuth
...
          */
-        cc7::U32 sessionIdentifier() const;
-        
+        std::string applicationKey() const;
+
```
Comment (kober32) [inline include/PowerAuth/Session.h:86]: Formatting

**Review by kober32 — CHANGES_REQUESTED** [review]
Looks OK (just one typo).

But it just seems like the change is not explained enough.

If there is legacy support for RN, why is a newer server needed?
Where (as a mobile developer) can I get the new configuration?
Why the change, what's the benefit (other than just having 1 config value)?

I'm pretty sure that a lot of developers will have these questions because they will upgrade the SDK (Android studio will tell them) and they won't understand what to do and if they should nag backend developers to upgrade.

**Comment by hvge** [comment]
This is just a small piece related to 1.8.0 release. The new server will be needed due to #529. I created the migration document in advance, because we know that this future release will require server 1.5.0 and newer to operate properly.

**Review by kober32 — APPROVED** [review]
👍🏻

**Comment by hvge** [comment]
I just fixed the grammar and the formatting in that header. We'll update the documentation as a part of [other tasks](https://github.com/wultra/powerauth-mobile-sdk/milestone/11) planned for 1.8. We declare the compatibility in the compatibility matrix for each branch of SDK, so If Android developer proactively upgrade to an incompatible version, then unfortunately it will not work. There's a new status endpoint planned for server 1.5, so we can introduce an error code indicating that mobile SDK is not compatible with the server.

**Pull request merged**